### ISSUE TITLE
[New build - EXPERIMENTAL] 21w01a: Holomap System Map

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#3452]
A new holomap mode is introduce beside the galaxy map: the System Map.
Use the system map to view planets, stations and spaceship belonging to the system you are currently located in.
Two viewing mode can be activated:
 - Map: planets are ordained and station are visible.
 - Orrery: planets are shown at their real position. Stations and spaceship are visible.
 
 Some tweak have been made to the planetary system generation: 
 you will have to migrate any entity built on the space habitat as it will spawn in a different place.

Bugfixes:
#3364 Holomap not showing stars.
#3393 Teleport power stats are incorrect.
#3395 Gravity unit does not draw power.
#3433 Hinge incorrect title in config window.
#3413 Crashes with very large ship.
#3380 Cannot load some world different from the starting one.
#3436 Space Habitat gravity is now extending higher.

**Comment by tsunamayo** [comment]
Build is live! Happy new year everyone! This one took a while, there was a lot of refactor behind the scene on the planetary system generation.

**Comment by Kaiser-Indrasil** [comment]
I can see you very much liked the Elite: Dangerous' system map, and for a good reason! ;)

I've got some remarks for the system map and system procedural generation that might be helpful in future iterations.

1. The 3d view, planet spawning and orbits

![image](https://user-images.githubusercontent.com/63756057/104013698-b6047380-51a9-11eb-8881-3bfd5c7834fb.png)

I love the inclined, eccentric orbits (that are much more extreme than these of Kerbal Space Program's!) but they don't look very convincing. I mean, the part where one gas giant's rings can intersect another's. Same goes for one of the right-hand giant's moons which seems to pass through its rings. If you're planning to ever introduce orbital revolving of celestial bodies, this would be a crucial thing to fix. But even if not (which I'd totally understand, as it would be quite an undertaking for a questionably low payoff), they would still look, if not more realistic, then at least convincing enough. Starship Evo's arcade setting, as opposed to a realistic one, lets it bend the rules much more - hence the very tight planet orbits - but I think setting them just ever so slightly apart so they would no longer intersect would help the immersion and make the setting look that much more convincing.

Also it would be really handy to be able to move the map around with the left click in case there are clusters of ships somewhere outside the sphere of influence of the system satellites.

If the station's and a ship's icons overlap, it's impossible to select the ship. The problem piles up if there are multiple ships near that station.

When I click on a planet, the camera focuses on it, but I can't zoom out quite far enough to see the whole star system.

2. The 2d chart of the star system map

![image](https://user-images.githubusercontent.com/63756057/104014487-06300580-51ab-11eb-8889-cc52bb072c99.png)

Latching onto that *immersion* argument from before, I can see the reason behind the naming scheme using simple numbers to categorise planets, but the established convention is to name the celestial bodies after their mother star's name, so for example, Drasioseus 4-3, if it's a third moon of the fourth planet.

Also, the star system names' first letter could really use some capitalisation.

When I click on a space station's icon, it would be handy if it's info bubble went away after I clicked on a celestial body or anywhere in empty space.

I think there's a bug where a star is provided with a station icon in both views of the star system map.

Also, I think it's unnecessary to have the orbits displayed in the 2d chart as it clutters up the view.

**Comment by tsunamayo** [comment]
Yes the orbit still needs some work, as for the name generation (I tried to spend a minimum amount of time on this). I dont think I will ever do orbital mechanic. The orbit in map view must be a bug, I didnt got that when I was working on it.
To select single ship I guess  we would need a tighter zoom, otherwise it would never work (dimension are too small). But I made sure you select the station first, because the player is interested in station first.
Yeah it is like Elite, but when you think of it there is just not many ways of doing this. I could have dropped the map view but I liked it... You get all the info at a glance!

**Comment by Kaiser-Indrasil** [comment]
May I suggest a possible solution to present the cluttered sections of the map?
Eve Online does it in quite a neat way. If there are multiple entities occupying the same space, the UI generates a vertical list when you hover over the cluster with your cursor.

Then, the list remains open as long as you hover your mouse over it or until you've selected the entity.

![image](https://user-images.githubusercontent.com/63756057/104081255-6499ca00-5225-11eb-8c44-47a12ea266c8.png)

**Comment by tsunamayo** [comment]
@Kaiser-Indrasil good idea, I will see if I can do the same! Thanks for the valuable feedback, I guess I didnt played enough games ;)
